src/shared/hooks/useIsMounted.ts

```diff
@@ -1,6 +1,21 @@
-import { useCallback } from 'react'
-import useIsMountedRef from '~/shared/hooks/useIsMountedRef'
-export default (): (() => boolean) => {
-    const isMountedRef = useIsMountedRef()
-    return useCallback(() => !!isMountedRef.current, [isMountedRef])
+import { useCallback, useEffect, useRef } from 'react'
+
+/**
+ * @deprecated `useIsMounted` will often lead to confusing
+ * behaviour. Alternatives:
+ * - in effects: use a local `mounted` flag and unset it on discard.
+ * - in callbacks: use `AbortController` and `AbortSignal`.
+ */
```

Comment:
Ok. I was sure I saw somewhere that you were replacing the implementation of useIsMounted and that it will help but I just played around with trying to provide a different implementation and yeah - it doesn't change anything. One thing that I'd add here though in the deprecated comment - an example of how to make the local mounted flag

Reply:
Yeah, there's something called `useInterrupt` which is, well, an attempt to solve the issue. It's decent but I wouldn't do it the same way. Too complicated.

I'll include your suggestion! 👍🏻